lib/types.ts

```diff
@@ -1,6 +1,7 @@
 export interface StaticAnalysisOptions {
   imagePath?: string;
   imageType?: ImageType;
+  tmpDirPath?: string;
```

Comment:
why is this ```undefined```-able?

Reply:
To better capture the intent that when the plugin is called, these options may not actually be set... There is no guarantee that they'll be defined unless we strictly check that

Reply:
but you strictly check that 😂

Reply:
this should communicate what's expected (this does not mean we shouldn't strictly check them in addition)

Reply:
When I first grab the options none of the values are set, that's why we do the undefined checks. I could better capture the intent by initially marking the type as Partial<StaticAnalysisOptions>, then after the checks we would know it's of type StaticAnalysisOptions, WDYT?

Reply:
I think it's alright?
not too familiar with TS conventions.
I think that as long as ```StaticAnalysisOptions``` is true to its expectations, it's fine

Reply:
Considering the path is now optional, I've not made any changes. Do you think it's good to go in its current state?

Reply:
I would raise the issue of the other two parameters being ```?``` but it's not part of this context.
so yes, I think it's great :)

Reply:
Noted 😄